src/main/java/com/celements/pagelayout/DefaultLayoutService.java

```diff
@@ -0,0 +1,488 @@
+    return HtmlDoctype.getHtmlDoctype(Strings.emptyToNull(layoutPropertyObj.getStringValue(
+        fieldName))).toJavaUtil();
+  }
+
+  final DocumentReference getPageLayoutPropertiesClassRef(WikiReference wikiRef) {
+    return RefBuilder.from(wikiRef)
+        .space(PAGE_LAYOUT_PROPERTIES_CLASS_SPACE)
+        .doc(PAGE_LAYOUT_PROPERTIES_CLASS_DOC)
+        .build(DocumentReference.class);
+  }
+
+  final String getDefaultLayout() {
+    String defaultLayout = modelContext.getXWikiContext().getWiki().Param(
+        XWIKICFG_CELEMENTS_LAYOUT_DEFAULT, SIMPLE_LAYOUT);
+    final String requestAction = modelContext.getXWikiContext().getAction();
+    if ((requestAction != null) && !"view".equals(requestAction)) {
+      defaultLayout = modelContext.getXWikiContext().getWiki()
+          .Param(XWIKICFG_CELEMENTS_LAYOUT_DEFAULT + "." + requestAction, defaultLayout);
+      LOGGER.debug("getDefaultLayout for action [{}] got [{}].", requestAction, defaultLayout);
+    } else {
+      LOGGER.debug("getDefaultLayout got [{}].", defaultLayout);
+    }
+    return defaultLayout;
+  }
+
+  @NotNull
+  final SpaceReference getCelLayoutEditorSpaceRef() {
+    return RefBuilder.from(modelContext.getWikiRef())
+        .space(CEL_LAYOUT_EDITOR_PL_NAME)
+        .build(SpaceReference.class);
+  }
+
+  final WikiReference getCentralWikiRef() {
+    return RefBuilder.create().wiki(CelConstant.CENTRAL_WIKI_NAME).build(WikiReference.class);
+  }
+
+  private String getDocType() {
+    return cfgSrc.getProperty("celements.layout.docType", HtmlDoctype.HTML5.getValue());
+  }
+
+  @SuppressWarnings("unchecked")
+  private Deque<SpaceReference> getRenderingLayoutStack() {
+    ExecutionContext execContext = execution.getContext();
+    if (execContext.getProperty(CEL_RENDERING_LAYOUT_STACK_PROPERTY) == null) {
+      Deque<SpaceReference> layoutStack = new LinkedList<>();
+      execContext.setProperty(CEL_RENDERING_LAYOUT_STACK_PROPERTY, layoutStack);
+    }
+    return (Deque<SpaceReference>) execContext.getProperty(CEL_RENDERING_LAYOUT_STACK_PROPERTY);
+  }
+
+  private void setRenderLayoutInVelocityContext(SpaceReference layoutSpaceRef) {
+    VelocityContext vcontext = (VelocityContext) modelContext.getXWikiContext().get("vcontext");
+    if (layoutSpaceRef != null) {
+      vcontext.put(CEL_RENDERING_LAYOUT_CONTEXT_PROPERTY,
+          References.cloneRef(layoutSpaceRef, SpaceReference.class));
+    } else {
+      vcontext.remove(CEL_RENDERING_LAYOUT_CONTEXT_PROPERTY);
+    }
+  }
+
+  /**
+   * For TESTS ONLY!!!
+   *
+   * @param injectedInheritorFactory
+   */
+  void inject_TEST_InheritorFactory(InheritorFactory injectedInheritorFactory) {
+    this.injectedInheritorFactory = injectedInheritorFactory;
+  }
+
+  private InheritorFactory getInheritorFactory() {
+    if (injectedInheritorFactory != null) {
+      return injectedInheritorFactory;
+    }
+    return new InheritorFactory();
+  }

```

Comment:
don't. instead add a default method overload for testing:
`SpaceReference getPageLayoutForDoc(DocumentReference documentReference, InheritorFactory inheritorFactory)`